packages/artillery/lib/platform/aws-lambda/index.js

```diff
@@ -744,6 +628,60 @@ class PlatformLambda {
     return lambdaRoleArn;
   }
 
+  async ensureECRImageExists() {
+    const ecr = new AWS.ECR({ apiVersion: '2015-09-21', region: this.region });
+
+    const params = {
+      repositoryName: this.ecrImageRepository,
+      imageIds: [{ imageTag: this.currentVersion }]
+    };
+
+    let data;
+    try {
+      data = await ecr.describeImages(params).promise();
+    } catch (err) {
+      if (err.code === 'RepositoryNotFoundException') {
+        throw new Error(`ECR repository not found: ${this.ecrImageRepository}`);
+      }
+      throw new Error(`ECR image not found: ${this.currentVersion}`);
+    }
+
+    if (data.imageDetails[0].imageTags.includes(this.currentVersion)) {
+      return;
+    } else {
+      throw new Error(`ECR image not found: ${this.currentVersion}`);
+    }
+  }
+
+  async checkIfNewLambdaIsNeeded({ memorySize, functionName }) {
+    const lambda = new AWS.Lambda({
+      apiVersion: '2015-03-31',
+      region: this.region
+    });
+
+    try {
+      const res = await lambda
+        .getFunctionConfiguration({
+          FunctionName: functionName
+        })
+        .promise();
+
+      if (res.MemorySize != memorySize) {
+        console.log(
```

Comment:
```suggestion
        debug(
```

Should be a debug log IMO - since function reuse is supposed to be transparent to the user.